DependencyInjection/Configuration.php

```diff
@@ -235,4 +225,62 @@ protected function configurePlugins(ArrayNodeDefinition $root)
             ->end()
         ->end();
     }
+
+    /**
+     * Add configuration for authentication plugin.
+     *
+     * @return ArrayNodeDefinition|\Symfony\Component\Config\Definition\Builder\NodeDefinition
+     */
+    private function addAuthenticationPluiginNode()
+    {
+        $builder = new TreeBuilder();
+        $node = $builder->root('authentication');
+        $node
+            ->useAttributeAsKey('name')
+            ->prototype('array')
+                ->validate()
+                    ->always()
+                    ->then(function ($config) {
+                        switch ($config['type']) {
+                            case 'basic':
+                                if (empty($config['username']) || empty($config['password'])) {
+                                    throw new InvalidConfigurationException('Authentication "basic" requires both "username" and "password".');
+                                }
```

Comment:
should we also complain about fields that should *not* be set? maybe too strict though.